### PR TITLE
fix: use regionalized sts calls for admin login

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes

configures a regional STS client when making calls for studio login. This is to support opt-in regions where non-regionalized STS tokens are not valid

#### Description of how you validated changes
Manually verified

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/utils/admin-helpers.ts [hunk 1]
```diff
@@ -93,7 +93,9 @@ async function getAdminCognitoCredentials(idToken: CognitoIdToken, identityId: s
 }
 
 async function getAdminStsCredentials(idToken: CognitoIdToken, region: string): Promise<AwsSdkConfig> {
-  const sts = new aws.STS();
+  const sts = new aws.STS({
+    stsRegionalEndpoints: 'regional',
+  });
```
Comment (sobolk) [inline packages/amplify-provider-awscloudformation/src/utils/admin-helpers.ts:98]: is sts available in all regions ?

**Review by jhockett — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/utils/admin-helpers.ts (hunk 1 above)
Comment (edwardfoyle) [reply packages/amplify-provider-awscloudformation/src/utils/admin-helpers.ts:98]: Looks like there are a handful of regions it's not available in, but it's in all regions that studio supports
https://docs.aws.amazon.com/general/latest/gr/sts.html

Comment (sobolk) [reply packages/amplify-provider-awscloudformation/src/utils/admin-helpers.ts:98]: Is that codepath running only in studio ? If yes then we're good. If not then would sdk fall back to global? or would we need a setting or fallback code ?

Comment (edwardfoyle) [reply packages/amplify-provider-awscloudformation/src/utils/admin-helpers.ts:98]: Yes this codepath only runs for studio enabled apps

**Review by sobolk — APPROVED** [review]

**Review by johnpc — APPROVED** [review]

**Pull request merged**